qlib/HttpServer.qm

```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
         listeners{id}.addHandlers(h);
     }
 
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, string handler_name) {
+        lm.enter();
+        on_exit lm.exit();
+
+        if (!listeners.hasKey(id)) {
+            throw "HTTP-SERVER-ERROR", sprintf("there is no listener with ID %d", id);
+        }
+        listeners{id}.removeHandler(handler_name);
+        HttpServer::HttpListener::stopIfNoHandlers(listeners{id});
+    }
+
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, HttpServer::AbstractHttpRequestHandler handler) {
```

Comment:
also here - change this method to return a bool if the listener was stopped or not